src/storage/src/hummock/compactor/fast_compactor_runner.rs

```diff
@@ -0,0 +1,556 @@
+
+                    self.executor
+                        .builder
+                        .add_raw_block(block, filter_data, smallest_key, largest_key, meta)
+                        .await?;
+                    self.executor.clear();
+                }
+                if !first.current_sstable().is_valid() {
+                    first.next_sstable().await?;
+                    continue;
+                }
+                first.init_block_iter().await?;
+            }
+
+            let target_key = second.current_sstable().key();
+            let iter = first.sstable_iter.as_mut().unwrap().iter.as_mut().unwrap();
+            self.executor.run(iter, target_key).await?;
+            if !iter.is_valid() {
+                first.sstable_iter.as_mut().unwrap().iter.take();
+                if !first.current_sstable().is_valid() {
+                    first.next_sstable().await?;
+                }
+            }
+        }
+        let rest_data = if !self.left.is_valid() {
+            if !self.right.is_valid() {
+                return self.executor.builder.finish().await;
+            }
+            &mut self.right
+        } else {
+            &mut self.left
+        };
+        {
+            let sstable_iter = rest_data.sstable_iter.as_mut().unwrap();
+            let target_key = FullKey::decode(&sstable_iter.sstable.value().meta.largest_key);
+            if let Some(iter) = sstable_iter.iter.as_mut() {
+                self.executor.run(iter, target_key).await?;
+                assert!(!iter.is_valid());
+            }
+            sstable_iter.iter.take();
+        }
+
+        while rest_data.is_valid() {
+            let mut sstable_iter = rest_data.sstable_iter.take().unwrap();
+            while sstable_iter.is_valid() {
+                let smallest_key = FullKey::decode(sstable_iter.next_block_smallest()).to_vec();
+                let (block, filter_data, block_meta) =
+                    sstable_iter.download_next_block().await?.unwrap();
+                skip_raw_block_count += 1;
+                skip_raw_block_size += block.len() as u64;
+                let largest_key = sstable_iter.current_block_largest();
+                self.executor
+                    .builder
+                    .add_raw_block(block, filter_data, smallest_key, largest_key, block_meta)
+                    .await?;
+            }
+            rest_data.next_sstable().await?;
+        }
+        let mut total_read_bytes = 0;
+        for sst in &self.left.sstables {
+            total_read_bytes += sst.file_size;
+        }
+        for sst in &self.right.sstables {
+            total_read_bytes += sst.file_size;
+        }
+        tracing::info!(
+            "OPTIMIZATION: skip {} blocks for task-{}, optimize {}% data compression",
+            skip_raw_block_count,
+            self.task_id,
+            skip_raw_block_size * 100 / total_read_bytes,
+        );
```

Comment:
I think it is better to expose a metric for this optimization so that we know whether it triggers without greping the logs.

Reply:
done